gaitmesh_ros/src/annotate_cloud_gaits_continuous.cpp

```diff
@@ -25,32 +25,41 @@ class ContinuousGaitMesh
 
         client_recast_ = nh_.serviceClient<recast_ros::InputMeshSrv>("/recast_node/input_mesh");
         cloud_mesh_sub_ = nh_.subscribe("/gaitmesh_ros/input_cloud_mesh", 1, &ContinuousGaitMesh::cloudMeshCallback, this);
+
+        ROS_INFO("Wait for Recast node update serve to come online...");
+        client_recast_.waitForExistence();
+        ROS_INFO("Recast node online!");
     }
 
     void processCloudMesh()
     {
-        pcl::PCLPointCloud2 pcl_pc2;
-        pcl::PolygonMesh mesh;
-        pcl_conversions::toPCL(cloud_mesh_.cloud, pcl_pc2);
-        pcl::PointCloud<pcl::PointXYZ>::Ptr input_cloud(new pcl::PointCloud<pcl::PointXYZ>);
-        pcl::fromPCLPointCloud2(pcl_pc2, *input_cloud);
-        pcl_conversions::toPCL(cloud_mesh_.mesh, mesh);
-
-        pcl::PointCloud<pcl::PointXYZ>::Ptr step2(new pcl::PointCloud<pcl::PointXYZ>());                    // outlier_removal_cloud
-        pcl::PointCloud<pcl::PointNormal>::Ptr step3(new pcl::PointCloud<pcl::PointNormal>());              // normal_cloud
-        pcl::PointCloud<pcl::PointXYZRGBNormal>::Ptr step4(new pcl::PointCloud<pcl::PointXYZRGBNormal>());  // gait_cloud
-        std::vector<char> mesh_labels = annotateCloud(input_cloud, mesh, radius_normals_, radius_robot_, curvature_threshold_, step2, step3, step4);
-
-        recast_ros::InputMeshSrv srv;
-        srv.request.reference_point = cloud_mesh_.reference_point;
-        srv.request.input_mesh = cloud_mesh_.mesh;
-        srv.request.area_labels.resize(mesh_labels.size());
-        for (size_t i = 0; i < mesh_labels.size(); i++)
-            srv.request.area_labels[i] = mesh_labels[i];
-        if (client_recast_.call(srv))
-            ROS_INFO("Map was sent");
-        else
-            ROS_ERROR("Failed to send map");
+        if (new_cloud_mesh_recieved_) {
```

Comment:
To have less diff here use `if (!new_cloud_mesh_recieved_) return;` - then you dont need the block to be change this much